openblas@0.3.23 build fails with gmake@4.4.1: ld: ../libopenblas_skylakex-r0.3.23.so: undefined reference to `dgetrf_single'

### Steps to reproduce the issue

`openblas@0.3.23` is failing build using:
* `spack@develop` (3ccc527d746444b9f13dbea685f8bb880e21140c from `Wed Jun 28 17:23:50 2023 -0500`)
* GNU Make 4.4.1
* GCC 11.1.0
* Ubuntu 20.04, x86_64

The problem seems to have been introduced as a result of
* https://github.com/spack/spack/pull/38591
* https://github.com/spack/spack/commit/51c75c6da31d803760a7235dbf285b55091222e4

Concretization:
```
$> spack spec -I openblas
Input spec
--------------------------------
 -   openblas

Concretized
--------------------------------
 -   openblas@0.3.23%gcc@11.1.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-ubuntu20.04-cascadelake
[+]      ^perl@5.36.0%gcc@11.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-cascadelake
[+]          ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-cascadelake
[+]          ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-cascadelake
[+]              ^diffutils@3.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-cascadelake
[+]                  ^libiconv@1.17%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-cascadelake
[+]          ^gdbm@1.23%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-cascadelake
[+]              ^readline@8.2%gcc@11.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-cascadelake
[+]                  ^ncurses@6.4%gcc@11.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-cascadelake
[+]                      ^pkgconf@1.9.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-cascadelake
[+]          ^zlib@1.2.13%gcc@11.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-cascadelake
```

Error:
```
$> spack install openblas
...
==> Installing openblas-0.3.23-cebezofwmpascso2mko6r4gqxampxod3
==> No binary for openblas-0.3.23-cebezofwmpascso2mko6r4gqxampxod3 found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/5d/5d9491d07168a5d00116cdc068a40022c3455bf9293c7cb86a65b1054d7e5114.tar.gz
==> No patches needed for openblas
==> openblas: Executing phase: 'edit'
==> openblas: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32' '-s' 'CC=/spack/lib/spack/env/gcc/gcc' 'FC=/spack/lib/spack/env/gcc/gfortran' 'MAKE_NB_JOBS=0' 'ARCH=x86_64' 'TARGET=SKYLAKEX' 'USE_LOCKING=1' 'USE_OPENMP=0' 'USE_THREAD=0' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'

7 errors found in build log:
     2578    ar: `u' modifier ignored since `D' is the default (see `U')
     2579    ar: `u' modifier ignored since `D' is the default (see `U')
     2580    ar: `u' modifier ignored since `D' is the default (see `U')
     2581    ar: `u' modifier ignored since `D' is the default (see `U')
     2582    ar: `u' modifier ignored since `D' is the default (see `U')
     2583    ar: `u' modifier ignored since `D' is the default (see `U')
  >> 2584    /usr/bin/ld: ../libopenblas_skylakex-r0.3.23.so: undefined reference to `cgetrf_single'
  >> 2585    /usr/bin/ld: ../libopenblas_skylakex-r0.3.23.so: undefined reference to `sgetrf_single'
  >> 2586    /usr/bin/ld: ../libopenblas_skylakex-r0.3.23.so: undefined reference to `zgetrf_single'
  >> 2587    /usr/bin/ld: ../libopenblas_skylakex-r0.3.23.so: undefined reference to `dgetrf_single'
  >> 2588    collect2: error: ld returned 1 exit status
  >> 2589    make[1]: *** [Makefile:196: ../libopenblas_skylakex-r0.3.23.so] Error 1
  >> 2590    make: *** [Makefile:132: shared] Error 2
```


### Error message

<details><summary>Error message</summary>
<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.21.0.dev0 (3ccc527d746444b9f13dbea685f8bb880e21140c)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/11901007/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11901008/spack-build-out.txt)

@sethrj @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

Also affecting macOS build of OpenBLAS:
* [openblas/a5kjxmx 0.3.23 apple-clang@=14.0.0 darwin-ventura-aarch64 Machine Learning MPS](https://gitlab.spack.io/spack/spack/-/jobs/7516395)
```
...
make[1]: *** [Makefile:155: libopenblas-r0.3.23.dylib] Error 1
make: *** [Makefile:141: shared] Error 2
==> Error: ProcessError: Command exited with status 2:
    'make' '-j12' '-s' 'CC=/Users/gitlab-runner-p-0/builds/p3qDz9zd/0/spack/spack/lib/spack/env/clang/clang' 'FC=/Users/gitlab-runner-p-0/builds/p3qDz9zd/0/spack/spack/lib/spack/env/clang/gfortran' 'MAKE_NB_JOBS=0' 'ARCH=arm64' 'DYNAMIC_ARCH=1' 'DYNAMIC_OLDER=1' 'TARGET=GENERIC' 'USE_LOCKING=1' 'USE_OPENMP=0' 'USE_THREAD=0' 'RANLIB=ranlib' 'libs' 'netlib' 'shared'
4 errors found in build log:
     5065                     ^
     5066    1 warning generated.
     5067    ../kernel/arm64/../generic/trsm_kernel_RN.c:182:18: warning: varia
             ble 'jj' set but not used [-Wunused-but-set-variable]
     5068      BLASLONG i, j, jj;
     5069                     ^
     5070    1 warning generated.
  >> 5071    Undefined symbols for architecture arm64:
     5072      "_LAPACKE_get_nancheck", referenced from:
     5073          _LAPACKE_sbbcsd in libopenblas-r0.3.23.a(lapacke_sbbcsd.o)
     5074          _LAPACKE_sbdsdc in libopenblas-r0.3.23.a(lapacke_sbdsdc.o)
     5075          _LAPACKE_sbdsqr in libopenblas-r0.3.23.a(lapacke_sbdsqr.o)
     5076          _LAPACKE_sbdsvdx in libopenblas-r0.3.23.a(lapacke_sbdsvdx.o)
     5077          _LAPACKE_sdisna in libopenblas-r0.3.23.a(lapacke_sdisna.o)
     ...
     5086          _gotoblas_corename in libopenblas-r0.3.23.a(dynamic_arm64.o)
     5087          _gotoblas_dynamic_init in libopenblas-r0.3.23.a(dynamic_arm6
             4.o)
     5088      "_gotoblas_NEOVERSEV1", referenced from:
     5089          _gotoblas_corename in libopenblas-r0.3.23.a(dynamic_arm64.o)
     5090          _gotoblas_dynamic_init in libopenblas-r0.3.23.a(dynamic_arm6
             4.o)
     5091    ld: symbol(s) not found for architecture arm64
  >> 5092    collect2: error: ld returned 1 exit status
  >> 5093    make[1]: *** [Makefile:155: libopenblas-r0.3.23.dylib] Error 1
  >> 5094    make: *** [Makefile:141: shared] Error 2
See build log for details:
```

--- a/var/spack/repos/builtin/packages/openblas/package.py
+++ b/var/spack/repos/builtin/packages/openblas/package.py
@@ -463,9 +463,7 @@ class Openblas(MakefilePackage):
 
     @property
     def build_targets(self):
-        # Do not build tests: see #38591
-        # And note that "shared" is allowed (but ignored) when ~shared
-        return ["-s"] + self.make_defs + ["libs", "netlib", "shared"]
+        return ["-s"] + self.make_defs + ["all"]
 
     @run_after("build")
     @on_package_attributes(run_tests=True)

